DQMOffline/Trigger/plugins/LepHTMonitor.cc

```diff
@@ -0,0 +1,446 @@
+#include "DQMOffline/Trigger/plugins/LepHTMonitor.h"
+
+#include <limits>
+#include <algorithm>
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/Common/interface/TriggerNames.h"
+
+#include "DataFormats/HLTReco/interface/TriggerObject.h"
+
+#include "DataFormats/Math/interface/deltaR.h"
+
+#include "DataFormats/EgammaCandidates/interface/ConversionFwd.h"
+#include "DataFormats/EgammaCandidates/interface/Conversion.h"
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "RecoEgamma/EgammaTools/src/ConversionTools.cc" //Would prefer to include header, but fails to find hasMatchedConversion definition without this..
+#include "DataFormats/PatCandidates/interface/Muon.h"
+
+#include "CommonTools/TriggerUtils/interface/GenericTriggerEventFlag.h"
+
+namespace{
+
+  //Offline electron definition
+  bool IsGood(const reco::GsfElectron &el, const reco::Vertex::Point &pv_position,
+              const reco::BeamSpot::Point &bs_position,
+              const edm::Handle<reco::ConversionCollection> &convs, double lep_counting_threshold){
+    const float dEtaIn = el.deltaEtaSuperClusterTrackAtVtx();
+    const float dPhiIn = el.deltaPhiSuperClusterTrackAtVtx();
+    const float sigmaietaieta = el.full5x5_sigmaIetaIeta();
+    const float hOverE = el.hcalOverEcal();
+    float d0 = 0.0, dz=0.0;
+    if(el.pt()<lep_counting_threshold) return false;
+    try{
+      d0=-(el.gsfTrack()->dxy(pv_position));
+      dz = el.gsfTrack()->dz(pv_position);
+    }catch(...){
+      edm::LogError("LepHTMonitor") << "Could not read electron.gsfTrack().\n";
+      return false;
+    }
+    float ooemoop = 1e30;
+    if(el.ecalEnergy()>0.0 && std::isfinite(el.ecalEnergy())){
+      ooemoop = fabs(1.0/el.ecalEnergy() - el.eSuperClusterOverP()/el.ecalEnergy());
+    }
+    const auto &iso = el.pfIsolationVariables();
+    const float absiso = iso.sumChargedHadronPt
+      + std::max(0.0, iso.sumNeutralHadronEt + iso.sumPhotonEt -0.5 * iso.sumPUPt);
+    const float relisowithdb = absiso/el.pt();
+
+    bool pass_conversion = false;
+    if(convs.isValid()){
+      try{
+        pass_conversion = !ConversionTools::hasMatchedConversion(el, convs, bs_position);
+      }catch(...){
+        edm::LogError("LepHTMonitor") << "Electron conversion matching failed.\n";
+        return false;
+      }
+    }
+
+    float etasc = 0.0;
+    try{
+      etasc = el.superCluster()->eta();
+    }catch(...){
+      edm::LogError("LepHTMonitor") << "Could not read electron.superCluster().\n";
+      return false;
+    }
+    if(fabs(etasc)>2.5){
```

Comment:
not sure what you mean. Anyway I wouldnt do a try catch when you can just check the validity of the reference.